### DISCUSSION TITLE
How to know which IP to pick from Servicelb

### DISCUSSION THREAD

**Discussion by wanclem** [k3s-io/k3s#11144] [Q&A] [answered]
Before now, I've always disabled the built in cloud controller manager that comes bundled with k3s.

But today I decided to leave it enabled as it is and created a service of type load balancer only to discover something strange.

While, my gateway controller traefik is deployed as `kind:Deployment` with a replica of just 1 I noticed that ServiceLb seems to be running as a DaemonSet as shown below:

```
kube-system    svclb-traefik-64427955-2p27k               2/2     Running   0               2m46s
kube-system    svclb-traefik-64427955-grl9s               2/2     Running   0               2m47s
kube-system    svclb-traefik-64427955-xzdz9               2/2     Running   0               2m47s
kube-system    svclb-traefik-64427955-xzrcw               2/2     Running   0               2m47s
kube-system    svclb-traefik-64427955-zjgcd               2/2     Running   0               2m46s
```

That leaves me with the question, which IP do I then create an A record for in my DNS settings for routing? All 5 IPs?

Is there no way to work with just a single IP, because I can imagine this scaling even more if I add an extra node to the cluster

**Comment by brandond** [comment, marked as answer]
Check the docs: https://docs.k3s.io/networking/networking-services#how-servicelb-works

It's not a real loadbalancer. As the docs say, it just allows you to successfully create Services of type LoadBalancer, and ensures that the IPs of nodes hosting the LoadBalancer pods are published via the service status field. You can point DNS entries at one or all of the node IPs.

If you want a real Virtual IP for your LoadBalancer services, you should disable ServiceLB and deploy something else.
